aptos-move/vm-genesis/src/lib.rs

```diff
@@ -308,35 +309,14 @@ fn create_and_initialize_validators(
     session: &mut SessionExt<impl MoveResolver>,
     validators: &[Validator],
 ) {
-    let mut owners = vec![];
-    let mut consensus_pubkeys = vec![];
-    let mut proof_of_possession = vec![];
-    let mut validator_network_addresses = vec![];
-    let mut full_node_network_addresses = vec![];
-    let mut staking_distribution = vec![];
-
-    for v in validators {
-        owners.push(MoveValue::Address(v.address));
-        consensus_pubkeys.push(MoveValue::vector_u8(v.consensus_pubkey.clone()));
-        proof_of_possession.push(MoveValue::vector_u8(v.proof_of_possession.clone()));
-        validator_network_addresses.push(MoveValue::vector_u8(v.network_addresses.clone()));
-        full_node_network_addresses
-            .push(MoveValue::vector_u8(v.full_node_network_addresses.clone()));
-        staking_distribution.push(MoveValue::U64(v.stake_amount));
-    }
     exec_function(
         session,
         GENESIS_MODULE_NAME,
         "create_initialize_validators",
         vec![],
         serialize_values(&vec![
             MoveValue::Signer(CORE_CODE_ADDRESS),
-            MoveValue::Vector(owners),
-            MoveValue::Vector(consensus_pubkeys),
-            MoveValue::Vector(proof_of_possession),
-            MoveValue::Vector(validator_network_addresses),
-            MoveValue::Vector(full_node_network_addresses),
-            MoveValue::Vector(staking_distribution),
+            MoveValue::vector_u8(bcs::to_bytes(validators).expect("Validators can be serialized")),
         ]),
     );
```

Comment:
Is this right?  it kinda seems like you should just be placing a bcs inside a bcs -- maybe...

Reply:
It works and is so much cleaner 🤷‍♂️